src/main/java/study/step2/domain/Lotto.java

```diff
@@ -0,0 +1,42 @@
+package study.step2.domain;
+
+import java.util.Objects;
+
+public class Lotto {
+
+  private LottoNumber lottoNumber;
```

Comment:
```suggestion
  private Set<LottoNumber> lottoNumbers;
```

이런식으로 사용할 수 있도록 구현해주세요!

Reply:
이렇게 되면 LottoNumber의 로직이 다 Lotto로 넘어와야하고, LottoNumber는 검증 로직만 가져야합니다.

그리고 LottoNumber에 대해 잘 생각해보면, 필요한 인스턴스는 딱 1 ~ 45까지 45개입니다. 그래서 캐싱을 이용해보면 좋지 않을까요?

보통 원시값 포장 객체의 경우 `Integer.valueOf()` `String.valueOf()` 처럼 `valueOf` 라는 메소드를 사용합니다.

그리고

`Integer.valueOf(1) == Integer.valueOf(1)`은 true가 됩니다. 즉, valueOf로 만든 인스턴스는 항상 똑같은 인스턴스입니다.

이러한 점을 고려해서 만들어보세요!